dsl/interpreter.py

```diff
@@ -0,0 +1,164 @@
+import os
+import re
+import logging
+
+from dsl.parser import Grammar
+
+logger = logging.getLogger('example_logger')
+logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
+DSL_EXTENSION = '.str'
+MODULES_DIR_NAME = 'dsl/modules'
+modules = {}
+def load_modules():
+    content = os.listdir(MODULES_DIR_NAME)
+    for file in content:
+        if os.path.isfile(os.path.join(MODULES_DIR_NAME, file)) and file.endswith('.py'):
+            name = file[:len(file)-3]
+            modules[name] = exec()
+
+
+def parse_file(input_name: str) -> str | None:
+    if input_name.endswith('.py'):
+        return input_name
+    if not input_name.endswith(DSL_EXTENSION):
+        logger.info(f'File {input_name} has incorrect extension')
+        return None
+    if not os.path.isfile(input_name):
+        logger.error(f'File {input_name} does not exist')
+    file = open(input_name, "r")
+    lines = file.readlines()
+    # print whole file to a string with removing comments
+    is_commented = False
+    data = ''
+    # gr = Grammar()
+    # clean from comments
+    for line in lines:
```

Comment:
Doesn't load the whole file into memory, reads it line by line instead (more memory efficient)
```suggestion
    # print whole file to a string with removing comments
    is_commented = False
    data = ''
    # gr = Grammar()
    # clean from comments
    for line in file:
```